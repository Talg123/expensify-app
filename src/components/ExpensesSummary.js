import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const expWord = expenseCount === 1 ? "expense" : "expenses";
    const total = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div>
            <h1> Viewing {expenseCount} {expWord} totaling { total }</h1>
        </div>
    );
};

const mapStateToProps = (state) =>{
    const visiable = selectExpenses(state.expenses,state.filters);
    return {
        expenseCount: visiable.length,
        expensesTotal: selectExpensesTotal(visiable)
    };
};
export default connect(mapStateToProps)(ExpensesSummary);