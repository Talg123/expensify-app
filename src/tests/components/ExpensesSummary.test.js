import React from 'react';
import { shallow } from 'enzyme';
import ExpensesSummary from '../../components/ExpensesSummary';

test('Should correctly render expenses summary with 1 expense',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={10} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expenses summary with multi expense',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expenseTotal={13330} />);
    expect(wrapper).toMatchSnapshot();
});