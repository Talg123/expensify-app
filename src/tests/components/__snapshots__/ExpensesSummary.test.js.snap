// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary with 1 expense 1`] = `
<div>
  <h1>
     Viewing 
    1
     
    expense
     totaling 
    $0.00
  </h1>
</div>
`;

exports[`Should correctly render expenses summary with multi expense 1`] = `
<div>
  <h1>
     Viewing 
    3
     
    expenses
     totaling 
    $0.00
  </h1>
</div>
`;
